codegen/smithy-aws-swift-codegen/src/main/kotlin/software/amazon/smithy/aws/swift/codegen/middleware/RetryMiddleware.kt

```diff
@@ -21,7 +27,9 @@ class RetryMiddleware : MiddlewareRenderable {
     override val position = MiddlewarePosition.AFTER
 
     override fun render(writer: SwiftWriter, op: OperationShape, operationStackName: String) {
-        writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N(${middlewareParamsString()}))", AWSClientRuntimeTypes.Core.RetryerMiddleware)
+        val output = MiddlewareShapeUtils.outputSymbol(symbolProvider, model, op)
+        val outputError = MiddlewareShapeUtils.outputErrorSymbol(op)
+        writer.write("$operationStackName.${middlewareStep.stringValue()}.intercept(position: ${position.stringValue()}, middleware: \$N<\$N, \$N>(${middlewareParamsString()}))", AWSClientRuntimeTypes.Core.RetryerMiddleware, output, outputError)
```

Comment:
It's interesting that we now need to pass these in -- is this because we are now throwing specific exceptions in the Retryer's implementation?

Reply:
previously error type was passed throughout the middleware, now since it doesn't have to be since everything is async throws the middlewares that throw the specific type of error have to know about it thus we have to pass it in as a generic.